src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java

```diff
@@ -58,8 +59,21 @@ public void onBlockPlaceExisting(BlockPlaceEvent e) {
         // This prevents Players from placing a block where another block already exists
         // While this can cause ghost blocks it also prevents them from replacing grass
         // or saplings etc...
-        if (BlockStorage.hasBlockInfo(e.getBlock())) {
-            e.setCancelled(true);
+        Block block = e.getBlock();
+
+        if (e.getBlockReplacedState().getType().isAir()) {
+            SlimefunItem sfItem = BlockStorage.check(block);
+
+            if (sfItem != null) {
+                for (ItemStack item : sfItem.getDrops()) {
+                    if (!item.getType().isAir()) {
```

Comment:
I probably worded that wrongly, the null check is still required here, I simply meant that you could replace `.isAir()` with a `Material.AIR` check.
But either works tbh.

```suggestion
                    if (item != null && !item.getType().isAir()) {
```